V3/controllers target instance

leefrank9527 opened the pull request:
Fixed TargetInstance related issues and integrated with H3 Agent and Store.

leefrank9527 committed:
1. All: Fixed requestmapping issues.
2. QualityReviewToolController, QualityReviewToolControllerAttribute, TreeToolController, TreeToolControllerAttribute: declared attributes in separated class. Because @Controller could not be initialed as Bean.

leefrank9527 committed:
1. Fixed requestmapping issue.
2. Fixed invalid number issue: new Integer(null string).

leefrank9527 committed:
Set the prefix marker of command fields, so the command fields start
with "_" could be parsed.

leefrank9527 committed:
Added static field:     public static final String PARAM_OID="targetInstanceId";

leefrank9527 committed:
1. harvestLogManager(): init restTemplateBuilder.
2. environmentWCT(): init environmament and WebApplicationContext.
3. Replaced TreeToolController,QualityReviewToolController with TreeToolControllerAttribute,QualityReviewToolControllerAttribute.

leefrank9527 committed:
1. Commented requestmapping lines.
2. targetInstanceTabConfig(): fixed the view name of TabConfig.

leefrank9527 committed:
Added configuration item to ignore the authentication of requests from H3 Agent.

leefrank9527 committed:
Fixed the form actioni URL issues.

leefrank9527 committed:
Added invalidation error messages.

leefrank9527 committed:
Added project version update logic. Automaticly update the
project.version value in application.properties according to the application version in build.gradle when building project.

leefrank9527 committed:
Fixed rules compiled issues.

leefrank9527 committed:
Added TargetInstance Id column in UI to differentiate TargetInstances
clearly.

leefrank9527 committed:
Changed the WebApplicationContext initial to ApplicationContext initial.

leefrank9527 committed:
Added the paths related to LogReader and AgentClients.

leefrank9527 committed:
Fixed to match the new ArcHarvestResult and HarvestResult classes.

leefrank9527 committed:
Fixed to match the new ApplicationContext mechanism.

leefrank9527 committed:
Fixed webpage browser issues.

hannakoppelaar reviewed (changes requested):
For me, this breaks a number of unit tests.

leefrank9527 committed:
Fixed TargetInstance controllers and integration with Store related
issues.

leefrank9527 committed:
1. Called AgentClient and DigitalAssetStoreClient according to new
AbstractRestClient constructure method.
2. Added new configurations items of scheme (protocol type of
Http/Https)

leefrank9527 committed:
Fixed DigitalAssetStoreFactoryImpl bean initial issues.

leefrank9527 committed:
Removed redundant importede classes.

leefrank9527 committed:
Fixed ProfileBasicCredentialsController/ProfileFormCredentialsController
and related subclasses, validators.

leefrank9527 committed:
Fixed controller issues: added request mapping.

obrienben commented:
@leefrank9527 Hi Frank, two issues I've found in testing.

1. The `Target Instance Summary` screen. Updating the profile overrides, do not save when revisiting the same screen and are not visible when viewing the profile tab under the Target. See v2 for expected behaviour.
2. The fields in the `Target Instance -> Display` tab are not saving

Once fixed, I think all your PRs are ready to be merged.

leefrank9527 committed:
Fixed the content in Display Tab can not be saved issue.

leefrank9527 commented:
@obrienben I fixed the 2 issues you mentioned:
Issue 1. The overrided profile wasn't saved in Target Instance Summary screen: fixed the cascade type of TargetInstance in Core module.
Issue 2. The contents in Display tab page  wasn't saved in Target Instance Screen: fixed the tab initial of  ServletConfig in WebApp module.

obrienben committed:
Adjust application ports to resemble old tomcat ports used, for now.
